Linux arm64 warning in GDNative code: 'sysv_abi' attribute directive ignored [-Wattributes]

<!-- Please search existing issues for potential duplicates before filing yours:
https://github.com/godotengine/godot/issues?q=is%3Aissue
-->

**Godot version: 3.2.2-stable**
<!-- Specify commit hash if using non-official build. -->


**OS/device including `version:` Manjaro ARM Linux (Linux 5.4.51-2-MANJARO-ARM (aarch64)) + Raspberry Pi 4 4 GB**
<!-- Specify GPU model, drivers, and the backend (GLES2, GLES3, Vulkan) if graphics-related. -->



**Issue description: "warning: 'sysv_abi' attribute directive ignored [-Wattributes]" warning appears while trying to compile Godot engine**
<!-- What happened, and what was expected. -->


**Steps to reproduce:**

In general I am trying to compile Godot for my Raspberry Pi. I do that because there is no official Godot version for my setup at the moment. The full procedure description you can find [here.](https://docs.godotengine.org/en/3.1/development/compiling/compiling_for_x11.html) 

1. Open terminal and run: "sudo pacman -S scons pkgconf gcc libxcursor libxinerama libxi libxrandr mesa glu alsa-lib pulseaudio \
    freetype2 yasm"

2. Download official Godot version from here: [3.2.2-stable](https://github.com/godotengine/godot/releases/tag/3.2.2-stable) 

3. Extract files to: "/home/superadmin/Godot/godot-3.2.2-stable/"

4. Open terminal and run: "cd /home/superadmin/Godot/godot-3.2.2-stable/"

5. Then run: "scons -j8 platform=x11 bits=64"

Actual results: 

1. The compile time is taking too long. By the time I open this bug it still compiling and it like this for over 1.5 hr already.
2. Countless warnings appear in the console:

"
      |                                  ^
modules/gdnative/include/gdnative/gdnative.h:281:111: warning: 'sysv_abi' attribute directive ignored [-Wattributes]
  281 | void GDAPI godot_print_error(const char *p_description, const char *p_function, const char *p_file, int p_line);
      |                                                                                                               ^
modules/gdnative/include/gdnative/gdnative.h:282:113: warning: 'sysv_abi' attribute directive ignored [-Wattributes]
  282 | void GDAPI godot_print_warning(const char *p_description, const char *p_function, const char *p_file, int p_line);
      |                                                                                                                 ^
modules/gdnative/include/gdnative/gdnative.h:283:53: warning: 'sysv_abi' attribute directive ignored [-Wattributes]
  283 | void GDAPI godot_print(const godot_string *p_message);
      |                                                     ^
modules/gdnative/include/gdnative/gdnative.h:287:64: warning: 'sysv_abi' attribute directive ignored [-Wattributes]
  287 | bool GDAPI godot_is_instance_valid(const godot_object *p_object);
      |                                                                ^
modules/gdnative/include/gdnative/gdnative.h:290:65: warning: 'sysv_abi' attribute directive ignored [-Wattributes]
  290 | void GDAPI *godot_get_class_tag(const godot_string_name *p_class);

"


**Minimal reproduction project:**
<!-- A small Godot project which reproduces the issue. Drag and drop a zip archive to upload it. -->

--- a/modules/gdnative/include/gdnative/gdnative.h
+++ b/modules/gdnative/include/gdnative/gdnative.h
@@ -37,20 +37,24 @@ extern "C" {
 
 #if defined(_WIN32) || defined(__ANDROID__)
 #define GDCALLINGCONV
-#define GDAPI GDCALLINGCONV
+
 #elif defined(__APPLE__)
 #include "TargetConditionals.h"
 #if TARGET_OS_IPHONE
 #define GDCALLINGCONV __attribute__((visibility("default")))
-#define GDAPI GDCALLINGCONV
 #elif TARGET_OS_MAC
 #define GDCALLINGCONV __attribute__((sysv_abi))
-#define GDAPI GDCALLINGCONV
 #endif
-#else // !_WIN32 && !__APPLE__
+
+#else // Linux/BSD/Web
+#if defined(__aarch64__)
+#define GDCALLINGCONV
+#else
 #define GDCALLINGCONV __attribute__((sysv_abi))
-#define GDAPI GDCALLINGCONV
 #endif
+#endif
+
+#define GDAPI GDCALLINGCONV
 
 // This is for libraries *using* the header, NOT GODOT EXPOSING STUFF!!
 #if !defined(GDN_EXPORT)

